Source/Scene/Cesium3DTileStyle.js

```diff
@@ -67,39 +78,30 @@ define([
         this._showShaderFunctionReady = false;
         this._pointSizeShaderFunctionReady = false;
 
-        var style = this;
-        if (typeof data === 'string') {
-            RequestScheduler.request(data, loadJson).then(function(styleJson) {
-                setup(style, styleJson);
-                style._readyPromise.resolve(style);
-            }).otherwise(function(error) {
-                style._readyPromise.reject(error);
-            });
+        var that = this;
+        var promise;
+        if (typeof style === 'string') {
+            promise = RequestScheduler.request(style, loadJson);
         } else {
-            setup(style, data);
-            style._readyPromise.resolve(style);
+            promise = when.resolve(style);
         }
+
+        promise.then(function(styleJson) {
+            setup(that, styleJson);
+            that._readyPromise.resolve(that);
```

Comment:
This is redundant, you can just assign `this._readyPromise = promise.then....` and avoid using the defer completely. I'll just submit a tweak myself so we can get this merged.